### PR TITLE
Fail fast and direct to readme if dotnet is not on path

### PR DESCRIPTION
**Pull request by druttka**

### CODE REVIEW

File: iotedgedev/dotnetmoduleprocessor.py [hunk 1]
```diff
@@ -8,6 +8,12 @@ def __init__(self, envvars, utility, output, module_dir):
         self.output = output
         self.module_dir = module_dir
         self.exe_dir = self.envvars.DOTNET_EXE_DIR
+       
+        # Fail fast if dotnet is not on path
```
Comment (druttka) [inline iotedgedev/dotnetmoduleprocessor.py]: Heads up: My dev env is having some trouble, so I wasn't able to test this E2E yet. I was able to test the idea in a reduced, isolated python example.

File: iotedgedev/dotnetmoduleprocessor.py [hunk 2]
```diff
@@ -8,6 +8,12 @@ def __init__(self, envvars, utility, output, module_dir):
         self.output = output
         self.module_dir = module_dir
         self.exe_dir = self.envvars.DOTNET_EXE_DIR
+       
+        # Fail fast if dotnet is not on path
+        try:
+            self.utility.exe_proc(["dotnet", "--version"])
+        except:
+            raise ChildProcessException("The .NET Core SDK is required to use the Azure IoT Edge Dev Tool. See the README for installation instructions.") from None
```
Comment (jongio) [inline iotedgedev/dotnetmoduleprocessor.py]: Can you add the https://aka.ms/iotedgedev URL to the message?  Thanks!

**Comment by druttka** [comment]
@jongio  Here is the output if you run `iotedgedev build` without `dotnet` on path. I haven't found a way to suppress the traceback, but see message at end of output. Not sure yet what a unit test around this would look like. LMK what you think.

```
==================================
======== BUILDING MODULES ========
==================================

BUILDING MODULE: modules\filtermodule
Traceback (most recent call last):
  File "C:\Program Files\Python36\Scripts\iotedgedev-script.py", line 11, in <module>
    load_entry_point('iotedgedev', 'console_scripts', 'iotedgedev')()
  File "c:\program files\python36\lib\site-packages\click\core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 697, in main
    rv = self.invoke(ctx)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\program files\python36\lib\site-packages\click\core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "c:\program files\python36\lib\site-packages\click\decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "e:\iotedgedev\iotedgedev\cli.py", line 117, in build
    ctx.invoke(modules, build=True, push=push, deploy=deploy)
  File "c:\program files\python36\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "e:\iotedgedev\iotedgedev\cli.py", line 457, in modules
    mod.build()
  File "e:\iotedgedev\iotedgedev\modules.py", line 19, in build
    self.build_push(no_push=True)
  File "e:\iotedgedev\iotedgedev\modules.py", line 40, in build_push
    mod_proc = ModulesProcessorFactory(self.envvars, self.utility, self.output, module_dir).get(module_json.language)
  File "e:\iotedgedev\iotedgedev\modulesprocessorfactory.py", line 18, in get
    return DotNetModuleProcessor(self.envvars, self.utility, self.output, self.module_dir)
  File "e:\iotedgedev\iotedgedev\dotnetmoduleprocessor.py", line 16, in __init__
    raise ChildProcessError("The .NET Core SDK is required to use the Azure IoT Edge Dev Tool. For installation instructions, see the README at https://aka.ms/iotedgedev.") from None
ChildProcessError: The .NET Core SDK is required to use the Azure IoT Edge Dev Tool. For installation instructions, see the README at https://aka.ms/iotedgedev.
``

**Comment by druttka** [comment]
@jongio  sys.exit instead of raising an error seems legit. Used self.output.error instead of print for the message which formats it as red in supported terminals.

==================================
======== BUILDING MODULES ========
==================================

BUILDING MODULE: modules\filtermodule
ERROR: The .NET Core SDK is required to use the Azure IoT Edge Dev Tool. For installation instructions, see the README at https://aka.ms/iotedgedev.

**Comment by jongio** [comment]
Hey @druttka - Can you change the message:

"The .NET Core SDK is required to build .NET modules with the Azure IoT Edge Dev Tool. For installation...."

**Comment by druttka** [comment]
@jongio are you happy with this now?

```
==================================
======== BUILDING MODULES ========
==================================

BUILDING MODULE: modules\filtermodule
ERROR: The .NET Core SDK is required to use the Azure IoT Edge Dev Tool. For installation instructions, see the README at https://aka.ms/iotedgedev.
```

**Comment by jongio** [comment]
@druttka Yes, merge if tox passes.

**Pull request merged**